execute/internal/cache/commit_report_cache.go

```diff
@@ -0,0 +1,190 @@
+package cache
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"time"
+
+	"github.com/patrickmn/go-cache"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+	"github.com/smartcontractkit/chainlink-common/pkg/types/query/primitives"
+
+	readerpkg "github.com/smartcontractkit/chainlink-ccip/pkg/reader"
+	"github.com/smartcontractkit/chainlink-ccip/pkg/types/ccipocr3"
+)
+
+// CommitReportsCache optimizes calls to CommitReportsGTETimestamp by caching reports
+// and only fetching new reports since the last successful query.
+type CommitReportsCache struct {
+	lggr               logger.Logger
+	reports            *cache.Cache // key string -> CommitPluginReportWithMeta
+	lastQueryTimestamp time.Time    // timestamp of the most recent query
+}
+
+// NewCommitReportsCache creates a new commit reports cache.
+// Cached items live for messageVisibilityInterval + EvictionGracePeriod, mirroring
+// the lifetime used in commitRootsCache for executed roots.
+func NewCommitReportsCache(lggr logger.Logger, messageVisibilityInterval time.Duration) *CommitReportsCache {
+	ttl := messageVisibilityInterval + EvictionGracePeriod
+	return &CommitReportsCache{
+		lggr:    lggr,
+		reports: cache.New(ttl, CleanupInterval),
+	}
+}
+
+// generateKey creates a unique key for a commit report
+func generateKey(report ccipocr3.CommitPluginReportWithMeta) string {
+	// We'll create a unique key for each report by combining relevant fields
+	// For each blessed merkle root
+	for _, mrc := range report.Report.BlessedMerkleRoots {
+		return fmt.Sprintf("%s_%s", mrc.ChainSel.String(), mrc.MerkleRoot.String())
+	}
```

Comment:
IIUC, this key should be "re-org resistant"? i.e if a re-org occurs on the dest chain, they key generated from the newly mined commit report should be equal to the key prior to the re-org?